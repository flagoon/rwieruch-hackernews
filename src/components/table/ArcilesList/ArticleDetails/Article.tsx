import * as React from 'react'
import styled from 'styled-components';

interface IArticle {
    url: string,
    children?: React.ReactNode
}

const Article = ({ url, children }: IArticle) => {
    return <StyledArticle data-url={url}>{children}</StyledArticle>
}

const StyledArticle = styled.div`
    display: flex;
    border: 1px solid #ccc;
    border-radius: 5px;
    padding: 5px;
    margin: 1rem;
    align-items: center;

    :hover {
        background-color: #ccc;
        border-color: #aaa;
        cursor: pointer;
    }
`;

export default Article;