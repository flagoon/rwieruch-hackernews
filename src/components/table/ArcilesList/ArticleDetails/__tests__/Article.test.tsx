import 'jest-dom/extend-expect'
import * as React from 'react';
import { render } from 'react-testing-library';

import Article from '../Article'

describe('test', () => {
    it('should', () => {
        const mockedChildren = <div>test</div>;
        const { getByText } = render(<Article url='test'>{mockedChildren}</Article>);
        expect(getByText('test')).toMatchSnapshot();
    })

})