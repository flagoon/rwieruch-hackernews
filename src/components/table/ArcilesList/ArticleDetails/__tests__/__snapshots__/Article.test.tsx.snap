// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should 1`] = `
<div>
  test
</div>
`;
